### DISCUSSION TITLE
Which key should be used as the RA_TLS_EPID_API_KEY?

### DISCUSSION THREAD

**Discussion by jogi343** [gramineproject/gramine#1361] [Q&A] [answered]
Hi All,

I subscribed to the SGX EPID attestation service from [here](https://api.portal.trustedservices.intel.com/EPID-attestation), and once signed in, I can see the `SPID`, `primary key` and `secondary key`. In the [secret provisioning](https://github.com/gramineproject/gramine/tree/master/CI-Examples/ra-tls-secret-prov#quick-start) example, I need to provide the `RA_TLS_EPID_API_KEY`. So which key should be provided as the RA_TLS_EPID_API_KEY? Is it the primary key or secondary?

Regards,

**Comment by kailun-qin** [comment, marked as answer]
Hi @jogi343!

The `RA_TLS_EPID_API_KEY` refers to the subscription key (in Section 2.1, SGX Attestation API Spec[1]), which is the "unique API key that Service Provider needs to use to authenticate itself to the service". Both the Primary and Secondary keys are subscription keys [2]. The design of two keys is basically for giving continuity to the service in case a key needs to be regenerated.

So you can use **either** your Primary key or Secondary key as the `RA_TLS_EPID_API_KEY`.

[1] https://www.intel.com/content/dam/develop/public/us/en/documents/sgx-attestation-api-spec.pdf
[2] https://www.intel.com/content/www/us/en/support/articles/000090019/software/intel-security-products.html

**Reply by jogi343** [reply]
Thank you @kailun-qin , secret provisioning is working.
